cubeviz/tools/collapse_cube.py

```diff
@@ -0,0 +1,359 @@
+        hb_desc.addWidget(self.widget_desc)
+
+        # Create data component label and input box
+        self.data_label = QLabel("Data:")
+        self.data_label.setFixedWidth(100)
+        self.data_label.setAlignment((Qt.AlignRight | Qt.AlignTop))
+        self.data_label.setFont(boldFont)
+
+        self.data_combobox = QComboBox()
+        self.data_combobox.addItems([str(x).strip() for x in self.data.component_ids()
+                                     if not x in self.data.coordinate_components])
+        self.data_combobox.setMinimumWidth(200)
+
+        hb_data = QHBoxLayout()
+        hb_data.addWidget(self.data_label)
+        hb_data.addWidget(self.data_combobox)
+
+        # Create operation label and input box
+        self.operation_label = QLabel("Operation:")
+        self.operation_label.setFixedWidth(100)
+        self.operation_label.setAlignment((Qt.AlignRight | Qt.AlignTop))
+        self.operation_label.setFont(boldFont)
+
+        self.operation_combobox = QComboBox()
+        self.operation_combobox.addItems(operations.keys())
+        self.operation_combobox.setMinimumWidth(200)
+
+        hb_operation = QHBoxLayout()
+        hb_operation.addWidget(self.operation_label)
+        hb_operation.addWidget(self.operation_combobox)
+
+        # Create region label and input box
+        self.region_label = QLabel("region:")
+        self.region_label.setFixedWidth(100)
+        self.region_label.setAlignment((Qt.AlignRight | Qt.AlignTop))
+        self.region_label.setFont(boldFont)
+
+        self.region_combobox = QComboBox()
+
+        # Get the Specviz regions and add them in to the Combo box
+        for roi in self.parent.specviz._widget.roi_bounds:
+            self.region_combobox.addItem("Specviz ROI ({:.3}, {:.3})".format(roi[0], roi[1]))
+
+        self.region_combobox.addItems(["Custom (Wavelengths)", "Custom (Indices)"])
+        self.region_combobox.setMinimumWidth(200)
+        self.region_combobox.currentIndexChanged.connect(self._region_selection_change)
+
+        hb_region = QHBoxLayout()
+        hb_region.addWidget(self.region_label)
+        hb_region.addWidget(self.region_combobox)
+
+        # Create error label
+        self.error_label = QLabel("")
+        self.error_label.setFixedWidth(100)
+
+        self.error_label_text = QLabel("")
+        self.error_label_text.setMinimumWidth(200)
+        self.error_label_text.setAlignment((Qt.AlignLeft | Qt.AlignTop))
+
+        hbl_error = QHBoxLayout()
+        hbl_error.addWidget(self.error_label)
+        hbl_error.addWidget(self.error_label_text)
+
+        # Create start label and input box
+        self.start_label = QLabel("Start:")
+        self.start_label.setFixedWidth(100)
+        self.start_label.setAlignment((Qt.AlignRight | Qt.AlignTop))
+        self.start_label.setFont(boldFont)
+
+        self.start_text = QLineEdit()
+        self.start_text.setMinimumWidth(200)
+        self.start_text.setAlignment((Qt.AlignLeft | Qt.AlignTop))
```

Comment:
It would be very useful to populate both this and `end_text` with reasonable defaults. Maybe by default it should span the entire range of wavelengths? Or the middle 90%? I'll leave it up to someone else to decide what 'reasonable' means.

Reply:
I am going to leave this for now and let them put it in. I think this will force the idea that using the specviz ROI might be the best method (??)